src/gestures-hooks.tsx

```diff
@@ -161,11 +183,31 @@ class GestureActions {
     );
   };
 
+  importGestures = (gestures: Partial<GestureData>[]) => {
+    const validGestures: GestureData[] = [];
+    const importedGestureIcons: MakeCodeIcon[] = gestures
+      .map((g) => g.icon as MakeCodeIcon)
+      .filter(Boolean);
+    gestures.forEach((g) => {
+      if (g.ID && g.name !== undefined && Array.isArray(g.recordings)) {
+        if (!g.icon) {
+          g.icon = this.getDefaultIcon([
+            ...validGestures.map((g) => g.icon),
+            ...importedGestureIcons,
+          ]);
+        }
+        validGestures.push(g as GestureData);
+      }
+    });
+    this.setGestures(validGestures);
+  };
+
   setGestures = (gestures: GestureData[], isRetrainNeeded: boolean = true) => {
+    this.setGestureState({ data: [] });
```

Comment:
I don't think this is needed since you are setting gesture again a couple lines down
```suggestion
```

Reply:
Agreed. Done here with some more simplification [8a07982](https://github.com/microbit-foundation/ml-trainer/pull/291/commits/8a0798206e0bec764a257a766b105d9014c7eb2c).